codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -238,82 +172,26 @@ public T getAndRemove(T name, T defaultValue) {
     @Override
     public List<T> getAll(T name) {
         checkNotNull(name, "name");
-        List<T> values = new ArrayList<T>(4);
-        int h = hashCodeGenerator.generateHashCode(name);
+
+        LinkedList<T> values = new LinkedList<T>();
```

Comment:
@buchgr - We need to be able to do `addFirst` though. My hunch is that this is more expensive with ArrayList.  It would be ideal to have a `List` that doesn't "resize/copy" and also allows to add to front....any ideas?